src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -217,6 +218,20 @@ protected virtual ResourceLocationAttribute GetResourceLocationAttribute(Assembl
             return assembly.GetCustomAttribute<ResourceLocationAttribute>();
         }
 
+        protected virtual RootNamespaceAttribute GetRootNamespaceAttribute(Assembly assembly)
```

Comment:
In what scenario would users (not our tests) override this method? Leaning toward `internal virtual` instead of exposing this publicly.

Reply:
I chose `protected virutal` for the same reasons I did for `GetResourceLocationAttribute` just above this. The compromise we reached then was to include a warning against using it in the remarks, which I'll add in my reaction commit.